iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/stack.inl

```diff
@@ -30,8 +30,9 @@ inline cxx::optional<T> stack<T, Capacity>::pop() noexcept
         return cxx::nullopt;
     }
 
-    // low level memory management with access to the topmost element on the untyped buffer;
-    // type safety is ensured by the template parameter
+    // AXIVION Next Construct CertC++-EXP36 : every entry of m_data is aligned to alignof(T)
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : low level memory management with access to the topmost element on
```

Comment:
AutosarC++19_03-A5.2.4: The justification is not precise. reinterpret cast is safe here if the size and alignment of each array element is guaranteed (see  other comments). This requires the change of the element type (which of course depends on T, otherwise it would not work).